WordPress/gutenberg
# Block Bindings: Remove the experimental flag

## What?
Following up on #56867 and #57742.

Removes the experimental flag for Block Bindings:
- Removes the experiment from the GB experiments page.
- Removes all the places where we check for the experiment
- Also remove one instance of unnecessary `gutenberg-partial-pattern-syncing` check

**TIP:** Hide the whitespace when looking at the diff.

## Testing Instructions
1. Register a custom field. E.g.:
   
   ```php
   register_meta(
	'post',
	'text_custom_field',
	array(
		'show_in_rest' => true,
		'single'       => true,
		'type'         => 'string',
		'default'	   => 'Custom Text',
       )
    );
   ```
1. Open a post or page.
2. Insert a Paragraph block
3. Open the Code Editor
4. Inside the Paragraph block, add the following snippet which adds a binding: 

   ```json
   {"metadata":{"bindings":{"content":{"source":{"name":"post_meta","attributes":{"value":"text_custom_field"}}}}}}
   ```

5. Open the frontend of the site and make sure that the `text_custom_field` is present as the content of the Paragraph.
6. **Also**, make sure that the "Block bindings" experiement is no longer present on the GB Experiments page.
